### PR TITLE
Add demos to testing

### LINKED ISSUE #60: Add demos to testing
Following #58.

### PR DESCRIPTION
**Pull request by jwallwork23**
Closes #60.

### CODE REVIEW

File: test/test_demos.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# Set environment variable to specify that tests are being run so that we can
+# cut down the length of the demos
+os.environ["ANIMATE_REGRESSION_TEST"] = "1"
+
+cwd = os.path.abspath(os.path.dirname(__file__))
+demo_dir = os.path.abspath(os.path.join(cwd, "..", "demos"))
+all_demos = glob.glob(os.path.join(demo_dir, "*.py"))
+
+
+@pytest.fixture(params=all_demos, ids=lambda x: os.path.basename(x))
```
Comment (stephankramer) [inline test/test_demos.py:21]: The way things are set up here seems to want to future proof for a situation where we have multiple subdirectories under demos/ but in that case I foresee you would not necessarily have unique, meaningful (base) filename (this is currently an issue in Thetis)

File: test/test_demos.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    for f in glob.glob(os.path.join(source, "*.msh")):
+        shutil.copy(f, str(tmpdir))
+
+    # Change working directory to temporary directory
+    monkeypatch.chdir(tmpdir)
+    subprocess.check_call([sys.executable, demo_file])
+
+    # Clean up plots
+    for ext in ("jpg", "pvd", "vtu"):
+        subprocess.check_call(["rm", "-f", f"{demo_name}*.{ext}"])
```
Comment (stephankramer) [inline test/test_demos.py:41]: So you're assuming a convention where all generated output files are named <basename>.jpg/etc? Sounds sensible but is already not true for the first demo - and isn't strictly followed in goalie either, e.g. burgers_oo.py has burgers-oo_ee.jpg
I'm generally a little wary of duplicating this sort of logic here, that's typically also in the Makefile.

**Review by stephankramer — APPROVED** [review]
Happy for this to go in - we can also revisit when we do actually have more complex/multi-directory demos

File: test/test_demos.py (hunk 1 above)
Comment (jwallwork23) [reply test/test_demos.py]: To be honest, this was copied from Goalie, which was probably copied from Thetis at some point! Tweaked in https://github.com/pyroteus/animate/pull/63/commits/95b9f3257e2766cfb5aca5d275ceb6af31ab72f5.

File: test/test_demos.py (hunk 2 above)
Comment (jwallwork23) [reply test/test_demos.py]: Ah good spot. Fixed in https://github.com/pyroteus/animate/pull/63/commits/6137960d692703273d6eb7295c7421455013915d.

**Pull request merged**